skfda/preprocessing/dim_reduction/feature_extraction/_per_class_feature_transformer.py

```diff
@@ -0,0 +1,162 @@
+        >>> y = y.values.codes
+        >>> X_train, X_test, y_train, y_test = train_test_split(
+        ...    X, y, test_size=0.25, stratify=y, random_state=0)
+
+        >>> from skfda.preprocessing.dim_reduction.feature_extraction
+        ... import PerClassFeatureTransformer
+
+        Then we will need to select a fda transformer, and so we will
+        use RecursiveMaximaHunting
+
+        >>> from skfda.preprocessing.dim_reduction.variable_selection
+        ... import RecursiveMaximaHunting
+
+        Finally we need to fit the data and transform it
+
+        >>> t.fit(X_train, y_train)
+        >>> x_transformed = t.transform(X_test)
+
+        x_transformed will be a vector with the transformed data
+    """
+
+    def __init__(
+        self,
+        transformer: TransformerMixin,
+    ) -> None:
+        self.transformer = transformer
+        self._validate_transformer()
+
+    def _validate_transformer(
+        self,
+    ) -> None:
+        """
+        Check that the transformer passed is\
+        scikit-learn-like and that uses target data in fit.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        if not (hasattr(self.transformer, "fit")
+                and hasattr(self.transformer, "fit_transform")
+                and hasattr(self.transformer, "transform")
+                ):
+
+            raise TypeError(
+                "Transformer should implement fit and "
+                "transform. " + str(self.transformer)
+                + " (type " + str(type(self.transformer)) + ")"
+                " doesn't",
+            )
+
+        tags = self.transformer._get_tags()
+
+        if not (tags['stateless'] and tags['requires_y']):
+            raise TypeError(
+                "Transformer should use target data in fit."
+                + str(self.transformer)
+                + " (type " + str(type(self.transformer)) + ")"
+                " doesn't",
+            )
+
+    def fit(
+        self,
+        X: T,
+        y: np.ndarray,
+    ) -> PerClassFeatureTransformer:
+        """
+        Fit the model on each class using X as\
+        training data and y as target values.
+
+        Args:
+            X: FDataGrid with the training data.
+            y: Target values of shape = (n_samples).
+
+        Returns:
+            self
+        """
+        classes, class_feature_transformers = _fit_feature_transformer(
+            X,
+            y,
+            self.transformer,
+        )
+
+        self._classes = classes
+        self._class_feature_transformers_ = class_feature_transformers
+
+        return self
+
+    def transform(self, X: T) -> np.ndarray:
+        """
+        Transform the provided data using the already fitted transformer.
+
+        Args:
+            X: FDataGrid with the test samples.
+
+        Returns:
+            Array of shape (n_samples, G).
+        """
+        sklearn_check_is_fitted(self)
+
+        return [
```

Comment:
A list is not the appropriate type. You should use the same approximation as in the `FeatureUnion`.

Reply:
Do you mean adding a tag requiring a ndarray and returning either the ndarray or a DataFrame as in FdaFeatureUnion?

Reply:
Great, I'll change it today and push it